lib/email_tracking_retriever.py

```diff
@@ -46,7 +46,7 @@ def mark_as_unread(self, email_id) -> None:
       mail = self.get_all_mail_folder()
       mail.uid('STORE', email_id, '-FLAGS', '(\Seen)')
 
-  def get_trackings(self) -> Dict[str, Tracking]:
+  def get_trackings(self, is_amazon) -> Dict[str, Tracking]:
```

Comment:
We have subclasses for the amazon and bby versions of this; we should use this fact. Instead of adding an extra boolean (and the resulting extra switching logic), just make the `log_in_if_necessary` method do nothing by default, then have AmazonTrackingRetriever override it to actually do the thing.

Reply:
Doing this seems to cause an issue where i think the driver for other calls is not created since it's in `AmazonTrackingRetriever`
```
Unexpected error fetching tracking from email ID 59996: RetryError: RetryError[<Future at 0x106ce8790 state=finished raised AttributeError>]: <class 'tenacity.RetryError'>
RetryError[<Future at 0x106ce8790 state=finished raised AttributeError>]
  File "/Users/jmizrahi/Documents/Buying/order-tracking/lib/email_tracking_retriever.py", line 76, in get_trackings
    success, new_trackings = self.get_trackings_from_email(email_id, mail, attempt)

  File "/Users/jmizrahi/Documents/Buying/order-tracking/lib/email_tracking_retriever.py", line 194, in get_trackings_from_email
    tracking_nums = self.get_tracking_numbers_from_email(email_str, from_email, to_email)

  File "/Users/jmizrahi/Documents/Buying/order-tracking/lib/amazon_tracking_retriever.py", line 90, in get_tracking_numbers_from_email
    attempt = self.get_tracking_info_logged_out(url, self.driver)

  File "/Users/jmizrahi/Documents/Buying/order-tracking/lib/amazon_tracking_retriever.py", line 154, in get_tracking_info_logged_out
    self.load_url(driver, amazon_url)

  File "/Users/jmizrahi/.pyenv/versions/3.7.4/lib/python3.7/site-packages/tenacity/__init__.py", line 329, in wrapped_f
    return self.call(f, *args, **kw)

  File "/Users/jmizrahi/.pyenv/versions/3.7.4/lib/python3.7/site-packages/tenacity/__init__.py", line 409, in call
    do = self.iter(retry_state=retry_state)

  File "/Users/jmizrahi/.pyenv/versions/3.7.4/lib/python3.7/site-packages/tenacity/__init__.py", line 369, in iter
    six.raise_from(retry_exc, fut.exception())

  File "<string>", line 3, in raise_from

```